### PR TITLE
v5

### PR DESCRIPTION
**Pull request by DoctorMcKay**
# [Migration Guide](https://github.com/DoctorMcKay/node-steam-user/wiki/v4-%E2%86%92-v5-migration-guide)

# Breaking

- Now requires Node.js v14 or later (up from v8)
- Removed deprecated methods
    - `setSentry()`
- Removed deprecated events
    - `sentry`
- Removed deprecated login key functionality
    - Removed logOn properties: `loginKey` and `rememberPassword`
- Removed legacy authentication

# Other Changes

- Added `refreshToken` event
- Added `renewRefreshTokens` option

# To Do

- [x] Remove `setSentry()`
- [x] Remove `sentry` event
- [x] Remove login key functionality
- [x] Remove legacy auth
- [x] Add jsdoc to logOn
- [x] Probably add some event to access refresh tokens
- [x] Refresh token renewal

### CODE REVIEW

**Comment by DoctorMcKay** [comment]
I do intend to move to TypeScript at some point, but it shouldn't be a breaking change so no need to rush it here.

Sadly I don't think it's going to help with the inheritance madness since I'm not aware of any way to split a class into multiple files using TS either.

v5 is kind of an expedited thing since protobufjs has a significant security vulnerability which is only fixed in protobufjs v7, and updating that dependency is a breaking change since it drops support for node 8-12.

**Comment by Sadzurami** [comment]
> Sadly I don't think it's going to help with the inheritance madness since I'm not aware of any way to split a class into multiple files using TS either.

Am i right, the only problem with inheritance - intellisense that not picking up on methods?
If it is true, "declare" in typescript can be useful to solve this problem.

**Comment by DoctorMcKay** [comment]
> Am i right, the only problem with inheritance - intellisense that not picking up on methods? If it is true, "declare" in typescript can be useful to solve this problem.

Perhaps that works, though I'd have to go back and change everything back to using the old prototype syntax, which I don't think is really desirable.

**Review by Sadzurami — APPROVED** [review]

File: components/07-web.js [hunk 1]
```diff
@@ -34,11 +34,7 @@ class SteamUserWeb extends SteamUserWebAPI {
...
 
-		const {LoginSession, EAuthTokenPlatformType} = require('steam-session');
-		const CMAuthTransport = require('./classes/CMAuthTransport.js');
-
-		let transport = new CMAuthTransport(this);
-		let session = new LoginSession(EAuthTokenPlatformType.SteamClient, {transport});
+		let session = this._getLoginSession();
 		session.refreshToken = this._logOnDetails.access_token;
 		session.getWebCookies().then((cookies) => {
 			if (!cookies.some(c => c.startsWith('sessionid='))) {
```
Comment (Sadzurami) [inline components/07-web.js:40]: since version 1.3.0 node-steam-session always return sessionid

Comment (Sadzurami) [inline components/07-web.js:40]: since version 1.3.0 node-steam-session always return sessionid

Comment (DoctorMcKay) [reply components/07-web.js:40]: Doesn't hurt anything to double check.

**Comment by Sadzurami** [comment]
Looks like all todo's complete.
When can we expect an update?

**Comment by DoctorMcKay** [comment]
> Looks like all todo's complete. When can we expect an update?

I want to test it a little more. If you want to install from the branch and test it yourself with your code, that would help.

**Comment by Sadzurami** [comment]
> > Looks like all todo's complete. When can we expect an update?
> 
> I want to test it a little more. If you want to install from the branch and test it yourself with your code, that would help.

I tested v5 a few days after your last comment.
Tested it on some projects with many instances at the same time.

Looks pretty stable, no issues found.
But I don't test new features like `renewRefreshTokens`.
Used version 5 only as drop-in replacement for version 4.

**Pull request merged**